src/lib.rs

```diff
@@ -609,12 +606,14 @@ impl CrawlerConfig {
     }
 
     pub fn set_client(mut self, client: reqwest::Client) -> Self {
-        self.client = Some(Arc::new(client));
+        self.client = Some(client);
         self
     }
 
-    pub fn with_shared_client(mut self, client: Arc<reqwest::Client>) -> Self {
-        self.client = Some(client);
+    /// *NOTE* [`reqwest::Client`] already uses Arc under the hood, so
+    /// it's preferable to just `clone` it and pass via [`Self::set_client`]
+    pub fn with_shared_client(mut self, client: std::sync::Arc<reqwest::Client>) -> Self {
```

Comment:
```suggestion
    #[deprecated( since = "0.2", message = "You do not have to wrap the Client it in a `Arc` to reuse it, because it already uses an `Arc` internally. Users should instead use `set_client`instead."]
    pub fn with_shared_client(mut self, client: std::sync::Arc<reqwest::Client>) -> Self {
```